packages/@cdktf/provider-generator/lib/get/generator/provider-generator.ts

```diff
@@ -102,6 +102,8 @@ export class TerraformProviderGenerator {
           namespacedResources[namespace] = [];
         }
         namespacedResources[namespace].push(resourceModel);
+      } else if (resourceModel.structsRequireNamespace) {
```

Comment:
I'm a bit confused by the naming `structsRequireNamespace` (I know it existed previously) but will this create a namespace or "just" a resource that lives in a directory instead of a single file? Will this then already be a namespace?

On a related note: What happens, if a Terraform provider gets updated and a resource then passes the limit of 400: Will this automatically namespace the resource (Can it then be imported from `/lib/some-resource`)? If this happens unexpectedly, will users experience broken imports after running `get` for a new provider resource? Or will this only add the opportunity for that single resource to now be imported from a `/lib/some-resource` path while still keeping the old way of importing that resource?

Reply:
>On a related note: What happens, if a Terraform provider gets updated and a resource then passes the limit of 400: Will this automatically namespace the resource (Can it then be imported from /lib/some-resource)? If this happens unexpectedly, will users experience broken imports after running get for a new provider resource? Or will this only add the opportunity for that single resource to now be imported from a /lib/some-resource path while still keeping the old way of importing that resource?

This would be transparent to the user and work automatically without a breaking change.

> I'm a bit confused by the naming structsRequireNamespace (I know it existed previously) but will this create a namespace or "just" a resource that lives in a directory instead of a single file? Will this then already be a namespace?

Yeah, the name is confusing, should adapt it. This will perform the struct sharding without namespacing